src/main/interface/msp.c

```diff
@@ -868,9 +872,32 @@ bool mspProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst)
 
     case MSP_NAME:
         {
-            const int nameLen = strlen(pilotConfig()->name);
-            for (int i = 0; i < nameLen; i++) {
-                sbufWriteU8(dst, pilotConfig()->name[i]);
+            // Show warning for DJI OSD instead of pilot name if osd_warning_enabled
+            if (osdWarnGetState(OSD_WARNING_DJI)) {
+                bool isUsbConnected = false;
+#ifdef USE_VCP
+                isUsbConnected = usbVcpIsConnected();
+#endif
+                if (!isUsbConnected) {
+                    const char *warning = pilotConfig()->warning;
+                    const int len = strlen(warning);
+                    for (int i = 0; i < len; i++) {
+                        // skip non printable chars
+                        if (isprint(warning[i])) {
+                            sbufWriteU8(dst, warning[i]);
+                        } else {
+                            sbufWriteU8(dst, 0);
+                        }
+                    }
+                    break;
+                }
+            }
+
+            // Show current pilot name
+            const char *name = pilotConfig()->name;
+            const int len = strlen(name);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5623733)